src/aws-fsx/ontap-file-system.ts

```diff
@@ -0,0 +1,514 @@
+      preferredSubnet
+    ) {
+      throw new Error("'preferredSubnet' can only be specified for multi-AZ file systems");
+    }
+    if (preferredSubnet && !vpcSubnets.includes(preferredSubnet)) {
+      throw new Error("'preferredSubnet' must be one of the specified 'vpcSubnets'");
+    }
+  }
+
+  private validateRouteTables(deploymentType: OntapDeploymentType, routeTables?: aws_ec2.IRouteTable[]): void {
+    if (routeTables == null || routeTables.length === 0) {
+      return;
+    }
+    if (deploymentType !== OntapDeploymentType.MULTI_AZ_1 && deploymentType !== OntapDeploymentType.MULTI_AZ_2) {
+      throw new Error("'routeTables' can only be specified for multi-AZ file systems");
+    }
+  }
+
+  private validateThroughputCapacity(
+    deploymentType: OntapDeploymentType,
+    throughputCapacity?: number,
+    throughputCapacityPerHaPair?: number,
+    haPair: number = 1,
+  ): void {
+    if (
+      Token.isUnresolved(throughputCapacity) ||
+      Token.isUnresolved(throughputCapacityPerHaPair) ||
+      Token.isUnresolved(haPair)
+    ) {
+      return;
+    }
+    if (throughputCapacity == null && throughputCapacityPerHaPair == null) {
+      return;
+    }
+    if (throughputCapacity != null && throughputCapacityPerHaPair != null) {
+      throw new Error("'throughputCapacity' and 'throughputCapacityPerHaPair' cannot be specified at the same time");
+    }
+    if (haPair <= 0 || !Number.isInteger(haPair)) {
+      throw new Error(`'haPair' must be a positive integer, got ${haPair}`);
+    }
+
+    // Calculate the throughput per HaPair and use it for validation,
+    // regardless of whether `throughputCapacity` or `throughputCapacityPerHaPair` is defined.
+    const throughputPerHaPair = throughputCapacityPerHaPair ?? throughputCapacity! / haPair;
+
+    const validValues: { [key in OntapDeploymentType]: number[] } = {
+      SINGLE_AZ_1: [128, 256, 512, 1024, 2048, 4096],
+      MULTI_AZ_1: [128, 256, 512, 1024, 2048, 4096],
+      SINGLE_AZ_2: [1536, 3072, 6144],
+      MULTI_AZ_2: [384, 768, 1536, 3072, 6144],
+    };
+
+    const validRange = validValues[deploymentType];
+    if (!validRange.includes(throughputPerHaPair)) {
+      throw new Error(
+        `'throughputCapacityPerHaPair' and 'throughputCapacity' / haPairs must be one of the following values for ${deploymentType}: ${validRange.join(', ')}`,
+      );
+    }
+  }
+
+  private validateStorageCapacity(haPairs: number = 1, storageCapacityGiB: number): void {
+    if (Token.isUnresolved(storageCapacityGiB) || Token.isUnresolved(haPairs)) {
+      return;
+    }
+    if (
+      !Number.isInteger(storageCapacityGiB) ||
+      storageCapacityGiB < 1024 * haPairs ||
+      storageCapacityGiB > Math.min(1_048_576, 524_288 * haPairs)
+    ) {
+      throw new Error(
+        `storageCapacityGiB must be an integer between ${1024 * haPairs} and ${Math.min(1_048_576, 524_288 * haPairs)}, got ${storageCapacityGiB}`,
+      );
```

Comment:
This validation is based on this [description](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-fsx-filesystem.html#cfn-fsx-filesystem-storagecapacity).